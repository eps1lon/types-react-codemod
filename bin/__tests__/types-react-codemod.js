import { describe, expect, test } from "@jest/globals";
import * as childProcess from "child_process";
import * as path from "path";
import { promisify } from "util";

describe("types-react-codemod", () => {
	const exec = promisify(childProcess.exec);
	const typesReactCodemodBin = path.join(
		__dirname,
		"../types-react-codemod.cjs"
	);
	function execTypesReactCodemod(args) {
		return exec(`${typesReactCodemodBin} ${args}`, {});
	}

	test("provides help", async () => {
    // TODO: toMatchInlineSnapshot fails with "Couldn't locate all inline snapshots."
		await expect(
			execTypesReactCodemod("--help")
		).resolves.toMatchSnapshot();
	});
});
