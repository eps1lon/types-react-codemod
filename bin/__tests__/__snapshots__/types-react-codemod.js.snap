// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`types-react-codemod provides help 1`] = `
Object {
  "stderr": "",
  "stdout": "types-react-codemod <codemod> <paths...>

Positionals:
  codemod  [string] [required] [choices: \\"context-any\\", \\"deprecated-react-type\\",
   \\"deprecated-sfc-element\\", \\"deprecated-sfc\\", \\"deprecated-stateless-component\\",
                   \\"implicit-children\\", \\"preset-18\\", \\"useCallback-implicit-any\\"]
  paths                                                      [string] [required]

Options:
  --version         Show version number                                [boolean]
  --help            Show help                                          [boolean]
  --dry                                               [boolean] [default: false]
  --ignore-pattern                      [string] [default: \\"**/node_modules/**\\"]
  --verbose                                           [boolean] [default: false]

Examples:
  types-react-codemod preset-18 ./          Ignores \`node_modules\` and \`build\`
  --ignore-pattern                          folders
  \\"**/{node_modules,build}/**\\"
",
}
`;
